lib/aes/aes.rb

```diff
@@ -84,7 +84,7 @@ def random_iv
   
     # Generate a random key
     def random_key(length=256)
-      _random_seed.unpack('H*')[0][0..((length/8)-1)]
+      _random_seed.unpack('H*')[0][0..((length/4)-1)]
```

Comment:
The key as generated by `_random_seed` is a hex string. In a hex string, each character encodes 4 bits.